pkg/catalogsourceconfig/handler.go

```diff
@@ -14,15 +14,18 @@ import (
 )
 
 func NewHandler(mgr manager.Manager) Handler {
+	cache := NewCache()
 	return &catalogsourceconfighandler{
 		client: mgr.GetClient(),
 		scheme: mgr.GetScheme(),
 		factory: &phaseReconcilerFactory{
 			reader: datastore.Cache,
 			client: mgr.GetClient(),
+			cache:  cache,
 		},
 		transitioner: phase.NewTransitioner(),
 		reader:       datastore.Cache,
+		cache:        cache,
```

Comment:
It is being used in two places.

Reply:
Right, because NewCache() actually creates the object. This is definitely just a nit, but is it possible then for us to use the same pattern for all of the fields here? More than one `client` field is set by calling mgr.GetClient() in line for example. I know it makes sense from an object composition perspective, but it just seems messy to me. Not going to hold this up for this though.

Reply:
I agree but that is outside the scope of this PR. I will handle that in a follow up PR.